src/alloc_filter.rs

```diff
@@ -0,0 +1,64 @@
+use tikv_jemalloc_sys::malloc_usable_size;
+
+/// Allocation filter which allows recording data only for specific
+/// allocations, lowering CPU load and reducing profile noise.
+pub trait AllocFilter {
+    /// Given a pointer to the allocated memory, returns `true` if
+    /// the allocations needs to be recorded by the profiler.
+    fn should_record(&self, ptr: *const u8) -> bool;
+}
+
+/// Filter based on Jemalloc allocation bin size.
+#[derive(Debug)]
+pub struct JemallocMultiBinFilter<const NUM_BINS: usize> {
```

Comment:
Somehow I missed that we can generalize over constants. Learned new thing today!